### PR TITLE
Changed Payment Mode to Testmode

### PR DESCRIPTION
**Pull request by Kam1ni**
## Description

Changed Mode string in the Payment struct to Testmode bool as specified by the mollie documentation: https://docs.mollie.com/reference/v2/payments-api/create-payment

Update: Also fixed testmode=true not added to the URL when in testing is true.

## Motivation and context

I had received errors from Mollie when I set Mode to any variable other then "".  This fixes the issue.

## How has this been tested?

I tested it in a private repo by manually settings Testmode to true on a payment object. I also made sure that if a mollie client is created with testing enabled that Testmode will be automatically set to true.

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

Go over all the following points, and put an `x` in all the boxes that apply.

Please, please, please, don't send your pull request until all of the boxes are ticked. Once your pull request is created, it will trigger a build on our continuous integration server to make sure your [tests and code style pass](https://help.github.com/articles/about-required-status-checks/).

- [ ] I have read the **[CONTRIBUTING](CONTRIBUTING.md)** document.
- [x] My pull request addresses exactly one patch/feature.
- [ ] I have created a branch for this patch/feature.
- [x] Each individual commit in the pull request is meaningful.
- [x] I have added tests to cover my changes.
- [ ] If my change requires a change to the documentation, I have updated it accordingly.

If you're unsure about any of these, don't hesitate to ask. We're here to help!

### CODE REVIEW

**Comment by VictorAvelar** [comment]
Hi @Kam1ni thanks for opening this PR, I will look into it. Will keep you posted about any question or clarification if required.

**Comment by VictorAvelar** [comment]
Hello @Kam1ni, 

I would like to understand your use case better, last night I went through other language implementation for this same endpoint, the thing here is that my assumption is that you are misusing the object.

Mode has type PaymentMode because as specified on the API reference this is the value mollie retrieves to indicate the environment in which the payment was created (specially useful when using org access tokens) thus the only possible values for Mode field are live/test.

Making a boolean will have the subsequent side effects:
1. Users relying on the Mode field to filter payments by environment will rely on the boolean being set, which is flaky because Mollie will set this value to live/test on their side once the payment is processed.
2. We will have a different implementation for Go(lang) that for all the other languages which IMHO is not good nor recommended. (example for [PHP](https://github.com/mollie/mollie-api-php/blob/master/src/Resources/Payment.php#L30) and [Node](https://github.com/mollie/mollie-api-node/blob/master/src/data/payments/data.ts#L15))
3. We will need to do a major release as the change will break the whole payment creation lifecycle.

My suggestion for your use case based on the assumption that you really need to pass the Mode field along, is to use one of the Provided constants for [Mode(s)](https://github.com/VictorAvelar/mollie-api-go/blob/25f44b6b4a1089c3875ff46efe3f94760b01ac05/mollie/common_types.go#L136).

If you don't pass this constant then your payment will be processed and the Mode field will be set by Mollie according to the token (for API tokens) or testmode query parameter (for Access tokens).

So basically I have 2 questions:
1. Why do you consider is appropriate to make the Mode field TestMode and change the type to boolean?
2. Can you give me a bit more context on your use case and why you need to pass the Mode value from your side instead of allowing Mollie to set it up for you?

**Comment by Kam1ni** [comment]
Mode is not a valid property of the payment object as specified by the API. Testmode is the right property. Passing mode to the API will result in an error. https://docs.mollie.com/reference/v2/payments-api/create-payment

Setting mode automatically to true in the requests when your client has test mode enabled is just a convenience.

But now that I'm looking at the api again I noticed something else. https://docs.mollie.com/reference/v2/payments-api/get-payment Mode can only be used for getting and not setting. In other words I should have left Mode and add Testmode. Instead of replacing it with Testmode.

**Comment by VictorAvelar** [comment]
I cannot find anywhere in the docs a place where it mentions that parameter, I also reviewed most of the official mollie implementations across different languages and I couldn't find the field/property in any of the payment objects.

Would you be able to paste a cURL representation of the failing request with anonymized data? That way I can try to reproduce it using my own credentials, because my assumption is that the field is ignored by the API.

I also use the client myself and never ran into that issue.

Thanks for the info provided so far.

**Comment by Kam1ni** [comment]
The parameter is mentioned here https://docs.mollie.com/reference/v2/payments-api/create-payment under access-token parameters.
image

I also mentioned before that the failing request is because I thought Mode was a writeable value. But it's in fact read-only. So no point in sending you that cURL representation. But Testmode is still specified by the API if you're using access tokens.

**Comment by VictorAvelar** [comment]
I think I managed to reproduced the error you are reporting, I will do some extra testing on my side and then I will integrate your pr. Thanks for reporting man!.
